ming/session.py

```diff
@@ -151,16 +153,23 @@ def _prep_save(self, doc, validate):
         return data
 
     @annotate_doc_failure
-    def save(self, doc, *args, **kwargs):
+    def save(self, doc, *args, **kwargs) -> bson.ObjectId:
         data = self._prep_save(doc, kwargs.pop('validate', True))
         if args:
-            values = {arg: data[arg] for arg in args}
-            result = self._impl(doc).update(
-                dict(_id=doc._id), {'$set': values}, **fix_write_concern(kwargs))
+            data = {arg: data[arg] for arg in args}
+            result = self._impl(doc).update_one(
+                dict(_id=doc._id), {'$set': data},
+                upsert=True, **fix_write_concern(kwargs)
```

Comment:
the old update() call didn't pass upsert=True, do we need it here?

Reply:
My thinking was that we probably don't ever want to "fail" on a save. We probably always want to just... do the save. An upsert guarantees that. At least that was my thinking. I wasn't necessarily thinking about how we're currently using this fn, more about conceptually how I would expect it to work as a user.

Reply:
After discussing further in person. I realized that we don't want to upsert a document that only has a partial number of fields. So yeah, i took out this upsert.